[
  {
    "number": 2007,
    "title": "\u5386\u53f2\u6570\u636e\u52a0\u8f7d\u4e86\uff0c\u4f46\u662f\u8fd8\u662f\u663e\u793a\u201c\u6210\u4ea4\u8bb0\u5f55\u4e3a\u7a7a\uff0c\u65e0\u6cd5\u8ba1\u7b97\u201d",
    "created_at": "2019-08-07T10:08:04Z",
    "closed_at": "2019-08-07T10:24:02Z",
    "labels": [
      "question"
    ],
    "url": "https://github.com/vnpy/vnpy/issues/2007",
    "body": "## \u73af\u5883\r\n\r\n* \u64cd\u4f5c\u7cfb\u7edf: \u5982Windows 10\r\n* Anaconda\u7248\u672c: vnstudio v2.0.6\r\n* vn.py\u7248\u672c: vnstudio v2.0.6\r\n\r\n## Issue\u7c7b\u578b\r\nQuestion\r\n\r\n## \u9884\u671f\u7a0b\u5e8f\u884c\u4e3a\r\n\u56de\u6d4b\u6210\u529f\r\n\r\n## \u5b9e\u9645\u7a0b\u5e8f\u884c\u4e3a\r\n\u56de\u6d4b\u4e0d\u6210\u529f\uff0c\u663e\u793a\u201c\u6210\u4ea4\u8bb0\u5f55\u4e3a\u7a7a\uff0c\u65e0\u6cd5\u8ba1\u7b97\u201d\r\n\r\n## \u91cd\u73b0\u6b65\u9aa4\r\n1. \u5b8c\u6574\u4ee3\u7801\u5982\u4e0b\uff1a\r\n```python\r\nfrom vnpy.app.cta_strategy.backtesting import BacktestingEngine, OptimizationSetting\r\nfrom vnpy.app.cta_strategy.strategies.atr_rsi_strategy import (\r\n    AtrRsiStrategy,\r\n)\r\nfrom vnpy.app.cta_strategy.strategies.double_ma_strategy import (\r\n    DoubleMaStrategy,\r\n)\r\n\r\nfrom datetime import datetime\r\n\r\n#%%\r\nengine = BacktestingEngine()\r\nengine.set_parameters(\r\n    vt_symbol=\"XBTUSD.BITMEX\", #\u672c\u5730\u4ee3\u7801\r\n    interval=\"1m\", # K\u7ebf\u5468\u671f\r\n    start=datetime(2019, 8, 1), #\u5f00\u59cb\u65e5\u671f\r\n    end=datetime(2019, 8, 30), #\u7ed3\u675f\u65e5\u671f\r\n    rate=0.3/1000000, #\u624b\u7eed\u8d39\u7387\r\n    slippage=0, #\u4ea4\u6613\u6ed1\u70b9\r\n    size=10, #\u5408\u7ea6\u4e58\u6570\r\n    pricetick=0.5, #\u4ef7\u683c\u8df3\u52a8\r\n    capital=1_000_000, #\u56de\u6d4b\u8d44\u91d1\r\n)\r\n# AtrRsiStrategy\r\n# engine.add_strategy(AtrRsiStrategy, {})\r\n# BollChannelStrategy\r\n# DoubleMaStrategy\r\nengine.add_strategy(DoubleMaStrategy, {\r\n    \"fast_window\":20,\r\n    \"slow_window\":80\r\n})\r\n# DualThrustStrategy\r\n# KingKeltnerStrategy\r\n# MultiTimeframeStrategy\r\n# TestStrategy\r\n# TurtleSignalStrategy\r\n\r\n#%%\r\nengine.load_data()\r\nengine.run_backtesting()\r\ndf = engine.calculate_result()\r\nengine.calculate_statistics()\r\nengine.show_chart()\r\n\r\nsetting = OptimizationSetting()\r\nsetting.set_target(\"sharpe_ratio\")\r\nsetting.add_parameter(\"atr_length\", 3, 39, 1)\r\nsetting.add_parameter(\"atr_ma_length\", 10, 30, 1)\r\n\r\n# engine.run_ga_optimization(setting)\r\n\r\n```\r\n2. \u76f4\u63a5\u8fd0\u884c\u4ee3\u7801\r\n```shell\r\npython backtesting.py\r\n```\r\n3. \u6700\u7ec8\u8f93\u51fa\u7ed3\u679c\uff1a\r\n```shell\r\n2019-08-07 17:59:17.406999      \u5f00\u59cb\u52a0\u8f7d\u5386\u53f2\u6570\u636e\r\n2019-08-07 17:59:18.126817      \u52a0\u8f7d\u8fdb\u5ea6\uff1a########## [100%]\r\n2019-08-07 17:59:18.126817      \u5386\u53f2\u6570\u636e\u52a0\u8f7d\u5b8c\u6210\uff0c\u6570\u636e\u91cf\uff1a8641\r\n2019-08-07 17:59:18.246799      \u7b56\u7565\u521d\u59cb\u5316\u5b8c\u6210\r\n2019-08-07 17:59:18.246799      \u5f00\u59cb\u56de\u653e\u5386\u53f2\u6570\u636e\r\n2019-08-07 17:59:18.256820      \u5386\u53f2\u6570\u636e\u56de\u653e\u7ed3\u675f\r\n2019-08-07 17:59:18.256820      \u5f00\u59cb\u8ba1\u7b97\u9010\u65e5\u76ef\u5e02\u76c8\u4e8f\r\n2019-08-07 17:59:18.256820      \u6210\u4ea4\u8bb0\u5f55\u4e3a\u7a7a\uff0c\u65e0\u6cd5\u8ba1\u7b97\r\n2019-08-07 17:59:18.256820      \u5f00\u59cb\u8ba1\u7b97\u7b56\u7565\u7edf\u8ba1\u6307\u6807\r\n2019-08-07 17:59:18.256820      ------------------------------\r\n2019-08-07 17:59:18.256820      \u9996\u4e2a\u4ea4\u6613\u65e5\uff1a\r\n2019-08-07 17:59:18.266823      \u6700\u540e\u4ea4\u6613\u65e5\uff1a\r\n2019-08-07 17:59:18.266823      \u603b\u4ea4\u6613\u65e5\uff1a      0\r\n2019-08-07 17:59:18.276828      \u76c8\u5229\u4ea4\u6613\u65e5\uff1a    0\r\n2019-08-07 17:59:18.276828      \u4e8f\u635f\u4ea4\u6613\u65e5\uff1a    0\r\n2019-08-07 17:59:18.276828      \u8d77\u59cb\u8d44\u91d1\uff1a      1,000,000.00\r\n2019-08-07 17:59:18.276828      \u7ed3\u675f\u8d44\u91d1\uff1a      0.00\r\n2019-08-07 17:59:18.276828      \u603b\u6536\u76ca\u7387\uff1a      0.00%\r\n2019-08-07 17:59:18.276828      \u5e74\u5316\u6536\u76ca\uff1a      0.00%\r\n2019-08-07 17:59:18.276828      \u6700\u5927\u56de\u64a4:       0.00\r\n2019-08-07 17:59:18.276828      \u767e\u5206\u6bd4\u6700\u5927\u56de\u64a4: 0.00%\r\n2019-08-07 17:59:18.276828      \u603b\u76c8\u4e8f\uff1a        0.00\r\n2019-08-07 17:59:18.276828      \u603b\u624b\u7eed\u8d39\uff1a      0.00\r\n2019-08-07 17:59:18.276828      \u603b\u6ed1\u70b9\uff1a        0.00\r\n2019-08-07 17:59:18.276828      \u603b\u6210\u4ea4\u91d1\u989d\uff1a    0.00\r\n2019-08-07 17:59:18.286829      \u603b\u6210\u4ea4\u7b14\u6570\uff1a    0\r\n2019-08-07 17:59:18.286829      \u65e5\u5747\u76c8\u4e8f\uff1a      0.00\r\n2019-08-07 17:59:18.296839      \u65e5\u5747\u624b\u7eed\u8d39\uff1a    0.00\r\n2019-08-07 17:59:18.296839      \u65e5\u5747\u6ed1\u70b9\uff1a      0.00\r\n2019-08-07 17:59:18.296839      \u65e5\u5747\u6210\u4ea4\u91d1\u989d\uff1a  0.00\r\n2019-08-07 17:59:18.296839      \u65e5\u5747\u6210\u4ea4\u7b14\u6570\uff1a  0\r\n2019-08-07 17:59:18.296839      \u65e5\u5747\u6536\u76ca\u7387\uff1a    0.00%\r\n2019-08-07 17:59:18.296839      \u6536\u76ca\u6807\u51c6\u5dee\uff1a    0.00%\r\n2019-08-07 17:59:18.296839      Sharpe Ratio\uff1a  0.00\r\n2019-08-07 17:59:18.296839      \u6536\u76ca\u56de\u64a4\u6bd4\uff1a    0.00\r\n```\r\n\u7a0b\u5e8f\u5e94\u8be5\u662f\u6b63\u786e\u6267\u884c\u4e86\uff0c\u4f46\u6ca1\u6709\u6210\u4ea4\u8bb0\u5f55\uff0c\u8bf7\u95ee\u662f\u4e0d\u662f\u54ea\u4e2a\u53c2\u6570\u8bbe\u7f6e\u9519\u8bef\uff1f\r\n\r\n",
    "comments_url": "https://api.github.com/repos/vnpy/vnpy/issues/2007/comments",
    "author": "watano",
    "comments": [
      {
        "user": "vnpy",
        "created_at": "2019-08-07T10:24:02Z",
        "body": "\u6570\u636e\u91cf\u4e0d\u591f\uff0c\u4e0d\u8db3\u4ee5\u8ba9\u7b56\u7565\u5b8c\u6210\u521d\u59cb\u5316\u8ba1\u7b97\uff0c\u8bf7\u51c6\u5907\u66f4\u591a\u7684\u6570\u636e\u7528\u4e8e\u56de\u6d4b"
      }
    ]
  },
  {
    "number": 1116,
    "title": "\u56de\u6d4bvnpy\u4f8b\u5b50Dustrategy\u51fa\u73b0\u9519\u8bef\uff0c\u5df2\u7ecf\u5728ctasetting\u91cc\u9762\u597d\u7b56\u7565\u540d\u79f0\u4e86",
    "created_at": "2018-09-16T06:59:30Z",
    "closed_at": "2018-09-17T01:23:58Z",
    "labels": [
      "question"
    ],
    "url": "https://github.com/vnpy/vnpy/issues/1116",
    "body": "## \u73af\u5883\r\n\r\n* \u64cd\u4f5c\u7cfb\u7edf: \u5982Windows 7\u6216\u8005Ubuntu 16.04\r\n* Anaconda\u7248\u672c: \u5982Anaconda 4.0.0 Python 2.7 32\u4f4d\r\n* vn.py\u7248\u672c: \u5982v1.7\u53d1\u884c\u7248\u6216\u8005dev branch 20170920\uff08\u4e0b\u8f7d\u65e5\u671f\uff09\r\n\r\n## Issue\u7c7b\u578b\r\n\u4e09\u9009\u4e00\uff1aBug/Enhancement/Question\r\n\r\n## \u9884\u671f\u7a0b\u5e8f\u884c\u4e3a\r\n\r\n\r\n## \u5b9e\u9645\u7a0b\u5e8f\u884c\u4e3a\r\n\r\n\r\n## \u91cd\u73b0\u6b65\u9aa4\r\n\r\n\u9488\u5bf9Bug\u7c7b\u578bIssue\uff0c\u8bf7\u63d0\u4f9b\u5177\u4f53\u91cd\u73b0\u6b65\u9aa4\u4ee5\u53ca\u62a5\u9519\u622a\u56fe\r\n\r\n   from vnpy.trader.app.ctaStrategy.strategy.strategyDualThrust import DuStrategy\r\n\r\n   ImportError: cannot import name DuStrategy\r\n",
    "comments_url": "https://api.github.com/repos/vnpy/vnpy/issues/1116/comments",
    "author": "wkpzwy",
    "comments": [
      {
        "user": "vnpy",
        "created_at": "2018-09-17T01:23:58Z",
        "body": "\u7b56\u7565\u540d\u5b57\u53eb\u505aDualThrustStrategy"
      }
    ]
  },
  {
    "number": 1017,
    "title": "FCOIN \u662f\u4e0d\u662f\u9650\u5236\u4e0b\u5355\u554a\uff0c\u62115\u79d2\u4e0b\u4e00\u6b21\u5355\uff0c\u4e0b\u4e8620\u6b21\u5de6\u53f3\u5c31\u4e0b\u4e0d\u4e86\u4e86\u3002",
    "created_at": "2018-08-03T11:48:49Z",
    "closed_at": "2018-08-04T13:50:27Z",
    "labels": [
      "question"
    ],
    "url": "https://github.com/vnpy/vnpy/issues/1017",
    "body": "## \u73af\u5883\r\n\r\n* \u64cd\u4f5c\u7cfb\u7edf: \u5982Windows 7\u6216\u8005Ubuntu 16.04\r\n* Anaconda\u7248\u672c: \u5982Anaconda 4.0.0 Python 2.7 32\u4f4d\r\n* vn.py\u7248\u672c: \u5982v1.7\u53d1\u884c\u7248\u6216\u8005dev branch 20170920\uff08\u4e0b\u8f7d\u65e5\u671f\uff09\r\n\r\n## Issue\u7c7b\u578b\r\n\u4e09\u9009\u4e00\uff1aBug/Enhancement/Question\r\n\r\n## \u9884\u671f\u7a0b\u5e8f\u884c\u4e3a\r\n\r\n\r\n## \u5b9e\u9645\u7a0b\u5e8f\u884c\u4e3a\r\n\r\n\r\n## \u91cd\u73b0\u6b65\u9aa4\r\n\r\n\u9488\u5bf9Bug\u7c7b\u578bIssue\uff0c\u8bf7\u63d0\u4f9b\u5177\u4f53\u91cd\u73b0\u6b65\u9aa4\u4ee5\u53ca\u62a5\u9519\u622a\u56fe\r\n\r\n",
    "comments_url": "https://api.github.com/repos/vnpy/vnpy/issues/1017/comments",
    "author": "sage-peng",
    "comments": [
      {
        "user": "vnpy",
        "created_at": "2018-08-04T13:50:27Z",
        "body": "FCOIN\u7684\u6d41\u63a7\u975e\u5e38\u4e25\uff0c\u5b98\u7f51\u53f7\u79f010\u79d2100\u6b21\uff0c\u5b9e\u9645\u53ef\u80fd10\u79d2\u53ea\u67092-3\u6b21\uff0c\u800c\u4e14\u9891\u7e41\u8d85\u8fc7\u4f1a\u88ab\u7981IP"
      }
    ]
  },
  {
    "number": 756,
    "title": "\u5b89\u88c5\u5b8c\u6210\u540e\u8fd0\u884crun.py\u62a5\u9519\uff0c1.7\u7248\u672c\u548c1.8\u7248\u672c\u90fd\u662f",
    "created_at": "2018-02-27T03:10:04Z",
    "closed_at": "2018-02-28T02:01:28Z",
    "labels": [
      "question"
    ],
    "url": "https://github.com/vnpy/vnpy/issues/756",
    "body": "## \u73af\u5883\r\n\r\n* \u64cd\u4f5c\u7cfb\u7edf: Ubuntu 16.04\r\n* vn.py\u7248\u672c: v1.7 && v1.8\r\n\r\n## Issue\u7c7b\u578b\r\n\u4e09\u9009\u4e00\uff1aQuestion\r\n\r\n## \u9884\u671f\u7a0b\u5e8f\u884c\u4e3a\r\n\r\n\r\n## \u5b9e\u9645\u7a0b\u5e8f\u884c\u4e3a\r\n\r\n\r\n## \u91cd\u73b0\u6b65\u9aa4\r\nTraceback (most recent call last):\r\n  File \"run.py\", line 19, in <module>\r\n    from vnpy.trader.gateway import (ctpGateway, oandaGateway, ibGateway, \r\n  File \"/home/future_test/anaconda2/lib/python2.7/site-packages/vnpy-1.7.0-py2.7.egg/vnpy/trader/gateway/oandaGateway/__init__.py\", line 4, in <module>\r\n    from oandaGateway import OandaGateway\r\n  File \"/home/future_test/anaconda2/lib/python2.7/site-packages/vnpy-1.7.0-py2.7.egg/vnpy/trader/gateway/oandaGateway/oandaGateway.py\", line 27, in <module>\r\n    from vnpy.api.oanda import OandaApi\r\n  File \"/home/future_test/anaconda2/lib/python2.7/site-packages/vnpy-1.7.0-py2.7.egg/vnpy/api/oanda/__init__.py\", line 3, in <module>\r\n    from vnoanda import OandaApi\r\n  File \"/home/future_test/anaconda2/lib/python2.7/site-packages/vnpy-1.7.0-py2.7.egg/vnpy/api/oanda/vnoanda.py\", line 4, in <module>\r\n    import requests\r\n  File \"/home/future_test/anaconda2/lib/python2.7/site-packages/requests/__init__.py\", line 52, in <module>\r\n    from .packages.urllib3.contrib import pyopenssl\r\n  File \"/home/future_test/anaconda2/lib/python2.7/site-packages/requests/packages/urllib3/contrib/pyopenssl.py\", line 47, in <module>\r\n    import OpenSSL.SSL\r\n  File \"/home/future_test/anaconda2/lib/python2.7/site-packages/OpenSSL/__init__.py\", line 8, in <module>\r\n    from OpenSSL import rand, crypto, SSL\r\n  File \"/home/future_test/anaconda2/lib/python2.7/site-packages/OpenSSL/crypto.py\", line 13, in <module>\r\n    from cryptography.hazmat.primitives.asymmetric import dsa, rsa\r\n  File \"/home/future_test/anaconda2/lib/python2.7/site-packages/cryptography/hazmat/primitives/asymmetric/rsa.py\", line 14, in <module>\r\n    from cryptography.hazmat.backends.interfaces import RSABackend\r\n  File \"/home/future_test/anaconda2/lib/python2.7/site-packages/cryptography/hazmat/backends/__init__.py\", line 7, in <module>\r\n    import pkg_resources\r\nzlib.error: Error -2 while decompressing data: inconsistent stream state\r\n\r\n",
    "comments_url": "https://api.github.com/repos/vnpy/vnpy/issues/756/comments",
    "author": "JohnHush",
    "comments": [
      {
        "user": "vnpy",
        "created_at": "2018-02-28T01:43:26Z",
        "body": "\u51fa\u9519\u7684\u662f\u8fd9\u4e2a\u5e93\uff1acryptography\r\n\u7136\u540e\u9519\u8bef\u4fe1\u606f\u4e3a\uff1azlib.error: Error -2 while decompressing data: inconsistent stream state\r\n\r\nvn.py\u4e2d\u5bfc\u81f4\u8fd9\u6bb5\u51fa\u9519\u7684\u4ee3\u7801\u662f\u4f60\u52a0\u8f7d\u4e86OANDA\u7684\u63a5\u53e3\uff0c\u5982\u679c\u4e0d\u7528\u7684\u8bdd\u4ecerun.py\u4e2d\u79fb\u9664\u5c31\u597d\r\n\u5982\u679c\u4e00\u5b9a\u8981\u7528\u5c31\u5f97\u81ea\u5df1\u67e5\u627e\u9519\u8bef\u4fe1\u606f\u7684\u539f\u56e0\u4e86\uff0c\u5efa\u8bae\u53ef\u4ee5\u76f4\u63a5\u5f04\u4e2a\u65b0\u7684Ubuntu 16.04\u91cd\u88c5Anaconda 4.0.0"
      }
    ]
  },
  {
    "number": 720,
    "title": "\u81ea\u5efa\u7b56\u7565\u6a21\u5757\u56de\u6d4b\u65e0\u6cd5\u5f15\u7528\uff1aNo module named strategyXXX",
    "created_at": "2018-01-30T16:24:11Z",
    "closed_at": "2018-01-31T01:16:18Z",
    "labels": [
      "question"
    ],
    "url": "https://github.com/vnpy/vnpy/issues/720",
    "body": "## \u73af\u5883\r\n\r\n* \u64cd\u4f5c\u7cfb\u7edf: Windows 10\r\n* Anaconda\u7248\u672c: Anaconda 4.0.0 Python 2.7 32\u4f4d\r\n* vn.py\u7248\u672c: v1.7.2\r\n\r\n## Issue\u7c7b\u578b\r\n\u4e09\u9009\u4e00\uff1aQuestion\r\n\r\n\u4f7f\u7528runBackTesting\u8fdb\u884c\u56de\u6d4b\uff0c\u5f15\u7528vnpy.trader.app.ctaStrategy.strategy\u4e0b\u539f\u6709\u7684\u7b56\u7565\u6587\u4ef6\uff0c\u53ef\u4ee5\u6b63\u5e38\u8fdb\u884c\u56de\u6d4b\u3002\r\n\u4f46\u662f\u5f15\u7528\u6211\u65b0\u5efa\u7684\u7b56\u7565\u6587\u4ef6\uff0c\u6bd4\u5982\uff1a\r\n\u6587\u4ef6\u540d\u4e3astrategyNew.py\uff0c\u5728runBackTesting\u4e2d\u5f15\u7528\uff1a\r\nfrom vnpy.trader.app.ctaStrategy.strategy.strategyNew import newStrategy\r\n\u8fd0\u884c\u540e\uff0c\u7cfb\u7edf\u5c31\u4f1a\u63d0\u793a\u9519\u8bef\uff1a\r\nNo module named strategyNew\r\n\r\n\u5f88\u5947\u602a\uff0c\u7167\u7406\u6765\u8bf4\u90fd\u662fvnpy.trader.app.ctaStrategy.strategy\u4e0b\u7684\u6a21\u5757\uff0c\u90fd\u4f7f\u7528\u7684\u7edd\u5bf9\u5f15\u7528\uff0c\u4e3a\u4f55\u6211\u65b0\u5efa\u7684\u6a21\u5757\u5c31\u4f1a\u63d0\u793a\u627e\u4e0d\u5230\uff0c\u8bf7\u6559\r\n\r\n",
    "comments_url": "https://api.github.com/repos/vnpy/vnpy/issues/720/comments",
    "author": "unlimitnight0",
    "comments": [
      {
        "user": "vnpy",
        "created_at": "2018-01-31T01:16:18Z",
        "body": "1. \u628a\u7b56\u7565\u6587\u4ef6\u653e\u5230\u5b89\u88c5\u5230site-packages\u76ee\u5f55\u4e2d\u7684vnpy\u76f8\u5173\u76ee\u5f55\u4e0b\uff0c\u800c\u4e0d\u662f\u4f60\u4e0b\u8f7d\u89e3\u538b\u540e\u7684\u76ee\u5f55\u4e0b\r\n2. \u56de\u6d4b\u7684\u8bdd\uff0c\u76f4\u63a5\u653e\u5230CtaBacktesting\u76ee\u5f55\u4e0b\uff0c\u7136\u540efrom strategyNew import newStrategy\u5c31\u884c"
      }
    ]
  },
  {
    "number": 393,
    "title": "\u8fd0\u884crun.py\u3000\u9047\u5230\u627e\u4e0d\u5230\u6307\u5b9a\u6a21\u5757",
    "created_at": "2017-07-08T03:39:47Z",
    "closed_at": "2017-07-10T01:24:57Z",
    "labels": [
      "question"
    ],
    "url": "https://github.com/vnpy/vnpy/issues/393",
    "body": "\r\n  File \"D:\\ProgramData\\Anaconda2\\lib\\site-packages\\vnpy\\api\\ctp\\__init__.py\", line 3, in <module>\r\n    from vnctpmd import MdApi\r\n\r\nImportError: DLL load failed: \u627e\u4e0d\u5230\u6307\u5b9a\u7684\u6a21\u5757\u3002",
    "comments_url": "https://api.github.com/repos/vnpy/vnpy/issues/393/comments",
    "author": "zhikook",
    "comments": [
      {
        "user": "zsluedem",
        "created_at": "2017-07-08T06:13:18Z",
        "body": "\u4ec0\u4e48\u7248\u672c\u7684\uff1f\r\nD:\\ProgramData\\Anaconda2\\lib\\site-packages\\vnpy\\api\\ctp\u91cc\u9762\u6709\u4ec0\u4e48\u6587\u4ef6"
      },
      {
        "user": "vnpy",
        "created_at": "2017-07-08T13:29:08Z",
        "body": "1. \u68c0\u67e5Anaconda\u7248\u672c\u662f\u5426\u4e3a4.0.0 Python 2.7 32\u4f4d\r\n2. \u68c0\u67e5vcredist x86 2013\u662f\u5426\u5b89\u88c5"
      },
      {
        "user": "zhikook",
        "created_at": "2017-07-08T14:37:02Z",
        "body": "@zsluedem thanks ,list:\r\n\r\n__init__.py\r\n__init__.pyc\r\nctp_data_type.py\r\nctp_data_type.pyc\r\nthostmduserapi.dll\r\nthostmduserapi.so\r\nthosttraderapi.dll\r\nthosttraderapi.so\r\nvnctpmd.pyd\r\nvnctptd.pyd"
      },
      {
        "user": "zhikook",
        "created_at": "2017-07-08T15:03:26Z",
        "body": "@vnpy \u8c22\u8c22\uff0cvcredist x86 2013\u5b89\u88c5\u7248\u672c\u51fa\u9519\uff0c\u6211\u60f3\u95ee\u9898\u5e94\u8be5\u662f\u8fd9\u4e2a\uff0c\u73b0\u5728\u8fd9\u4e2a\u95ee\u9898\u6ca1\u6709\u4e86\uff0c\u4e0d\u8fc7\u9047\u5230\u65b0\u95ee\u9898\uff0c\u6211\u6b63\u5728\u5904\u7406\u3002\r\n\r\n  File \"D:\\ProgramData\\Anaconda2\\lib\\anydbm.py\", line 85, in open\r\n    return mod.open(file, flag, mode)\r\n\r\n  File \"D:\\ProgramData\\Anaconda2\\lib\\dbhash.py\", line 18, in open\r\n    return bsddb.hashopen(file, flag, mode)\r\n\r\n  File \"D:\\ProgramData\\Anaconda2\\lib\\bsddb\\__init__.py\", line 364, in hashopen\r\n    d.open(file, db.DB_HASH, flags, mode)\r\n\r\nDBNoSuchFileError: (2, 'No such file or directory')\r\n\r\n"
      },
      {
        "user": "zhikook",
        "created_at": "2017-07-08T15:06:18Z",
        "body": " #----------------------------------------------------------------------\r\n    def loadContracts(self):\r\n        \"\"\"\u4ece\u786c\u76d8\u8bfb\u53d6\u5408\u7ea6\u5bf9\u8c61\"\"\"\r\n        f = shelve.open(self.contractFileName)\r\n        if 'data' in f:\r\n            d = f['data']\r\n            for key, value in d.items():\r\n                self.contractDict[key] = value\r\n        f.close()\r\n        \r\n\u8bf7\u95ee\u662f\u5426\u8981\u5bf9\u90a3\u4e2a\u6587\u4ef6\u8bbe\u7f6e\u8bfb\u5199\u6743\u9650\uff0c\u6211\u7684\u64cd\u4f5c\u7cfb\u7edf\u662f\uff57\uff49\uff4e\uff44\uff4f\uff57\uff53\uff17\u3000\uff13\uff12"
      },
      {
        "user": "vnpy",
        "created_at": "2017-07-09T02:06:31Z",
        "body": "\u9700\u8981\u7ba1\u7406\u5458\u6743\u9650\uff0c\u7528\u7ba1\u7406\u5458\u8d26\u53f7\u6765\u542f\u52a8Python"
      },
      {
        "user": "Hilyme",
        "created_at": "2019-01-09T05:40:24Z",
        "body": "\u6211\u7684\u7cfb\u7edf\u662fwin10\uff0cconda\u662f4.3.30\u7684\uff0cpython\u662f2.7\u7684\uff0c\u7136\u540evc\u662f2015\u7684\uff0c\u8fd8\u662ffrom .vnctpmd import MdApi\r\nImportError: DLL load failed: %1 \u4e0d\u662f\u6709\u6548\u7684 Win32 \u5e94\u7528\u7a0b\u5e8f  \u8fd9\u4e2a\u95ee\u9898\uff0c\u8fd9\u662f\u4ec0\u4e48\u539f\u56e0"
      }
    ]
  },
  {
    "number": 2308,
    "title": "windows\u4e0b \u4f7f\u7528\u591a\u8fdb\u7a0b\u53c2\u6570\u4f18\u5316BUG",
    "created_at": "2020-01-19T17:32:32Z",
    "closed_at": "2020-01-20T17:07:31Z",
    "labels": [
      "help wanted"
    ],
    "url": "https://github.com/vnpy/vnpy/issues/2308",
    "body": "## \u73af\u5883\r\n\r\n* \u64cd\u4f5c\u7cfb\u7edf: Windows 10\r\n* Python\u7248\u672c: Python 3.7.4\r\n* vn.py\u7248\u672c: v2.0.9\r\n\r\n## Issue\u7c7b\u578b\r\n\u4e09\u9009\u4e00\uff1aBug\r\n\r\n## \u9884\u671f\u7a0b\u5e8f\u884c\u4e3a\r\n\r\n\r\n## \u5b9e\u9645\u7a0b\u5e8f\u884c\u4e3a\r\n\r\n\r\n## \u91cd\u73b0\u6b65\u9aa4\r\n2020-01-20 01:22:51.361774\t\u5f00\u59cb\u56de\u653e\u5386\u53f2\u6570\u636e\r\n2020-01-20 01:22:51.361774\t\u9010\u65e5\u76ef\u5e02\u76c8\u4e8f\u8ba1\u7b97\u5b8c\u6210\r\n2020-01-20 01:22:51.361774\t\u5f00\u59cb\u8ba1\u7b97\u7b56\u7565\u7edf\u8ba1\u6307\u6807\r\n2020-01-20 01:22:51.380722\t\u5386\u53f2\u6570\u636e\u56de\u653e\u7ed3\u675f\r\n2020-01-20 01:22:51.380722\t\u5f00\u59cb\u8ba1\u7b97\u9010\u65e5\u76ef\u5e02\u76c8\u4e8f\r\n2020-01-20 01:22:51.386706\t\u9010\u65e5\u76ef\u5e02\u76c8\u4e8f\u8ba1\u7b97\u5b8c\u6210\r\n2020-01-20 01:22:51.386706\t\u5f00\u59cb\u8ba1\u7b97\u7b56\u7565\u7edf\u8ba1\u6307\u6807\r\nException in thread Thread-4:\r\nmultiprocessing.pool.RemoteTraceback: \r\n\"\"\"\r\nTraceback (most recent call last):\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\site-packages\\talib\\__init__.py\", line 20, in wrapper\r\n    for arg in chain(args, kwargs.values())\r\nStopIteration\r\n\r\nDuring handling of the above exception, another exception occurred:\r\n\r\nTraceback (most recent call last):\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\multiprocessing\\pool.py\", line 121, in worker\r\n    result = (True, func(*args, **kwds))\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\site-packages\\vnpy\\app\\cta_strategy\\backtesting.py\", line 1210, in optimize\r\n    engine.run_backtesting()\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\site-packages\\vnpy\\app\\cta_strategy\\backtesting.py\", line 294, in run_backtesting\r\n    func(data)\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\site-packages\\vnpy\\app\\cta_strategy\\backtesting.py\", line 724, in new_bar\r\n    self.strategy.on_bar(bar)\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\site-packages\\vnpy\\app\\cta_strategy\\strategies\\double_ma_strategy.py\", line 73, in on_bar\r\n    fast_ma = am.sma(self.fast_window, array=True)\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\site-packages\\vnpy\\trader\\utility.py\", line 377, in sma\r\n    result = talib.SMA(self.close, n)\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\site-packages\\talib\\__init__.py\", line 24, in wrapper\r\n    return func(*args, **kwargs)\r\n  File \"_func.pxi\", line 9954, in talib._ta_lib.SMA\r\n  File \"_common.pxi\", line 30, in talib._ta_lib._ta_check_success\r\nException: TA_SMA function failed with error code 2: Bad Parameter (TA_BAD_PARAM)\r\n\"\"\"\r\n\r\nThe above exception was the direct cause of the following exception:\r\n\r\nTraceback (most recent call last):\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\threading.py\", line 926, in _bootstrap_inner\r\n    self.run()\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\threading.py\", line 870, in run\r\n    self._target(*self._args, **self._kwargs)\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\site-packages\\vnpy\\app\\cta_backtester\\engine.py\", line 295, in run_optimization\r\n    output=False\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\site-packages\\vnpy\\app\\cta_strategy\\backtesting.py\", line 569, in run_optimization\r\n    result_values = [result.get() for result in results]\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\site-packages\\vnpy\\app\\cta_strategy\\backtesting.py\", line 569, in <listcomp>\r\n    result_values = [result.get() for result in results]\r\n  File \"G:\\ProgramData\\Anaconda3\\lib\\multiprocessing\\pool.py\", line 657, in get\r\n    raise self._value\r\nException: TA_SMA function failed with error code 2: Bad Parameter (TA_BAD_PARAM)\r\n\u56e0\u4e3a\u56de\u6d4b\u7684\u662f\u7684\u53cc\u5747\u7ebf\u7b56\u7565\uff0c\u63d0\u793aTA_SMA\u9519\u8bef\uff0c\u81ea\u5df1\u6363\u9f13\u4e86\u534a\u5929\u6ca1\u5f04\u51fa\u6765\u3002\r\n\r\n\r\n",
    "comments_url": "https://api.github.com/repos/vnpy/vnpy/issues/2308/comments",
    "author": "qmzhn",
    "comments": [
      {
        "user": "vnpy",
        "created_at": "2020-01-20T14:05:11Z",
        "body": "Exception: TA_SMA function failed with error code 2: Bad Parameter\r\n\r\n\u8fd9\u91cc\u7684\u62a5\u9519\uff0c\u6211\u731c\u6d4b\u53ef\u80fd\u662f\u7b2c\u4e8c\u4e2a\u53c2\u6570n\uff0c\u4f18\u5316\u65f6\u51fa\u73b0\u4e860\u6216\u8005\u8d1f\u6570"
      },
      {
        "user": "qmzhn",
        "created_at": "2020-01-20T17:07:31Z",
        "body": "\u7ecf\u6653\u4f18\u5927\u795e\u63d0\u70b9\uff0c\u6d4b\u8bd5\u540e\u53d1\u73b0TA_SMA\u7684n\u4e0d\u80fd\u4e3a1\u2026\u2026"
      }
    ]
  },
  {
    "number": 762,
    "title": "ctaStrategy\u6a21\u5757\u65e0\u6cd5\u52a0\u8f7d\u5f53\u524d\u76ee\u5f55\u4e0b\u5b50\u76ee\u5f55\u7684\u7b56\u7565\u6587\u4ef6",
    "created_at": "2018-03-01T05:18:21Z",
    "closed_at": "2018-05-07T15:06:41Z",
    "labels": [
      "help wanted"
    ],
    "url": "https://github.com/vnpy/vnpy/issues/762",
    "body": "## \u73af\u5883\r\n\r\n* \u64cd\u4f5c\u7cfb\u7edf: Windows10\r\n* Anaconda\u7248\u672c: \u5982Anaconda 4.0.0 Python 2.7 32\u4f4d\r\n* vn.py\u7248\u672c: 1.73\r\n\r\n## Issue\u7c7b\u578b\r\n\u4e09\u9009\u4e00\uff1aBug\r\n\r\n## \u9884\u671f\u7a0b\u5e8f\u884c\u4e3a\r\n\u5728\u5f53\u524d\u5de5\u4f5c\u76ee\u5f55\u4e2d\uff0c\u7b56\u7565\u6587\u4ef6\u653e\u7f6e\u5728\u5f53\u524d\u5de5\u4f5c\u76ee\u5f55\u4e2d\u7684\u5b50\u76ee\u5f55\u3002\r\n## \u5b9e\u9645\u7a0b\u5e8f\u884c\u4e3a\r\napp\\ctaStratgy\\strategy\\__init__.py\u7684loadStrategyModule\u4f1a\u62a5\u9519\uff0c\u65e0\u6cd5\u52a8\u6001\u52a0\u8f7d\u5b50\u76ee\u5f55\u4e2d\u7684\u7b56\u7565\u6a21\u5757\u3002\r\n\r\n## \u91cd\u73b0\u6b65\u9aa4\r\n1.  \u5f53\u524d\u76ee\u5f55\u7ed3\u6784\uff1aproject\\strategy\\strategyAtrRsi.py\r\n2.  \u8fd0\u884cproject\u6839\u76ee\u5f55\u4e0b\u7684run.py\r\n3. \u4f1a\u65e0\u6cd5\u52a0\u8f7dstrategyAtrRsi.py\r\n\r\n\u4e2a\u4eba\u89e3\u51b3\u65b9\u6848\u662f\uff0c\u8bf7\u6307\u6b63\uff1a\r\nworkingPath = os.getcwd()\r\nfor root, subdirs, files in os.walk(workingPath,False):\r\n    for name in files:\r\n        # \u53ea\u6709\u6587\u4ef6\u540d\u4e2d\u5305\u542bstrategy\u4e14\u975e.pyc\u7684\u6587\u4ef6\uff0c\u624d\u662f\u7b56\u7565\u6587\u4ef6\r\n        if 'strategy' in name and '.pyc' not in name:\r\n            # \u6a21\u5757\u540d\u79f0\u65e0\u9700\u524d\u7f00\r\n            moduleName = name.replace('.py', '')\r\n            import sys\r\n            sys.path.append(root)\r\n            loadStrategyModule(moduleName) ",
    "comments_url": "https://api.github.com/repos/vnpy/vnpy/issues/762/comments",
    "author": "yesheng1984",
    "comments": [
      {
        "user": "vnpy",
        "created_at": "2018-03-02T01:40:31Z",
        "body": "\u8bbe\u8ba1\u4e0a\u5c31\u4e0d\u652f\u6301\u5de5\u4f5c\u76ee\u5f55\u4e0b\u5b50\u76ee\u5f55\u4e2d\u7684\u7b56\u7565\u6587\u4ef6\u52a0\u8f7d\u3002\r\n\r\n\u4fee\u6539\u65b9\u6cd5\uff1a\r\n1. strategy\u76ee\u5f55\u4e2d\u6dfb\u52a0__init__.py\r\n2. moduleName = 'strategy.' + moduleName\r\n\r\n\u8bf7\u8bd5\u8bd5\u770b\u5427"
      },
      {
        "user": "justinchou",
        "created_at": "2019-02-14T02:31:51Z",
        "body": "\u6211\u5728 MacOS\u4e0b\u5f15\u5165 `vnpy.trader.app.ctaStrategy` \u62a5\u9519\u63d0\u793a\uff1a\r\n```\r\n--------------------\r\nFailed to import strategy file strategyAtrRsi:\r\nTraceback (most recent call last):\r\n  File \"/anaconda3/envs/Python27/lib/python2.7/site-packages/vnpy/trader/app/ctaStrategy/strategy/__init__.py\", line 20, in loadStrategyModule\r\n    module = importlib.import_module(moduleName)\r\n  File \"/anaconda3/envs/Python27/lib/python2.7/importlib/__init__.py\", line 37, in import_module\r\n    __import__(name)\r\nImportError: No module named strategyAtrRsi\r\n--------------------\r\nFailed to import strategy file strategyBollChannel:\r\nTraceback (most recent call last):\r\n  File \"/anaconda3/envs/Python27/lib/python2.7/site-packages/vnpy/trader/app/ctaStrategy/strategy/__init__.py\", line 20, in loadStrategyModule\r\n    module = importlib.import_module(moduleName)\r\n  File \"/anaconda3/envs/Python27/lib/python2.7/importlib/__init__.py\", line 37, in import_module\r\n    __import__(name)\r\nImportError: No module named strategyBollChannel\r\n--------------------\r\nFailed to import strategy file strategyDoubleMa:\r\nTraceback (most recent call last):\r\n  File \"/anaconda3/envs/Python27/lib/python2.7/site-packages/vnpy/trader/app/ctaStrategy/strategy/__init__.py\", line 20, in loadStrategyModule\r\n    module = importlib.import_module(moduleName)\r\n  File \"/anaconda3/envs/Python27/lib/python2.7/importlib/__init__.py\", line 37, in import_module\r\n    __import__(name)\r\nImportError: No module named strategyDoubleMa\r\n--------------------\r\nFailed to import strategy file strategyMultiTimeframe:\r\nTraceback (most recent call last):\r\n  File \"/anaconda3/envs/Python27/lib/python2.7/site-packages/vnpy/trader/app/ctaStrategy/strategy/__init__.py\", line 20, in loadStrategyModule\r\n    module = importlib.import_module(moduleName)\r\n........\r\n```"
      }
    ]
  }
]